### ISSUE TITLE
Loading large data-list

### ISSUE THREAD

**Issue by gabydevdev** [LeaVerou/awesomplete#9442]
Hi,
I've been looking for autocomplete solutions for an extension I'm working, what it does is to give you an address, with state and city with just the zip code. Would this extension work with a large list of data (more than 100,000)?

**Comment by LeaVerou** [comment]
I don’t think it has been tested with that kind of dataset but I would love to find out how it performs and perhaps optimize!
I would recommend supplying the list as an array, that’s the most low-level way to do it, so that you avoid all possible overhead.
Please let me know, and if it’s too slow, we can figure out ways to optimize it :)

**Comment by ImBobby** [comment]
I did a quick test to cover 100,000 datalist. This was the setup:

``` HTML
<input type="text">
```

``` Javascript
var input = document.querySelector('input');
var datalist = [];

for (var i = 1; i <= 100000; i++) {
    datalist.push( i.toString() );
}

new Awesomplete( input, {
    list: datalist
});
```

On Chrome 40.0.2214.115 and Safari 8.0.3 Awesomplete immediately return datalist as I typed.

But when tested in Firefox 36, it was slightly slower than Chrome and Safari.

**Comment by LeaVerou** [comment]
Thanks so much for the data!
How much slower was Firefox? Did you measure, by any chance?

**Comment by gabydevdev** [comment]
Hi, I did some test with the data from my tables... it seems to work fine with the large data-list (that would be all the zip codes from Mexico country), not slow at all. But I had to use the data-list attribute 'cause it wasn't working making an array in JS.

I did have some trouble with the filter function that only takes the first characters that match what you're typing, understanding how you would call awesomplate from JS and also I need for the data-list in the other fields to update after adding the zip code, which i kind of do except that when I start typing nothing appears & the sugestions that appear are the ones from the old data list.
